File: src/main/java/destiny/astrology/classical/rules/debilities/Sunbeam.java
/**
 * @author smallufo 
 * Created on 2007/12/31 at 上午 3:25:34
 */ 
package destiny.astrology.classical.rules.debilities;

import java.util.Locale;

import destiny.astrology.HoroscopeContext;
import destiny.astrology.Planet;

/** Under the Sunbeans (between 8.5 and 17 from Sol). */
public final class Sunbeam extends Rule
{

  public Sunbeam()
  {
    super("Sunbeam");
  }

  @Override
  public boolean isApplicable(Planet planet, HoroscopeContext horoscopeContext)
  {
    if (planet != Planet.SUN)
    {
      if (horoscopeContext.getHoroscope().getAngle(planet , Planet.SUN) > 8.5 &&
          horoscopeContext.getHoroscope().getAngle(planet , Planet.SUN) <= 17)
      {
        addComment(Locale.TAIWAN , planet + " 被太陽曬傷 (Sunbeam)");
      }
    }
    return false;
  }

}

File: src/main/java/destiny/astrology/classical/rules/debilities/Sunbeam.kt
/**
 * @author smallufo
 * Created on 2007/12/31 at 上午 3:25:34
 */
package destiny.astrology.classical.rules.debilities

import destiny.astrology.Horoscope
import destiny.astrology.Planet
import destiny.astrology.Planet.SUN
import org.jooq.lambda.tuple.Tuple2
import java.util.*

/** Under the Sunbeams (between 8.5 and 17 from Sol).  */
class Sunbeam : Rule() {

  override fun getResult(planet: Planet, h: Horoscope): Optional<Tuple2<String, Array<Any>>> {
    return getResult2(planet , h).toOld()
  }

  override fun getResult2(planet: Planet, h: Horoscope): Pair<String, Array<Any>>? {
    return planet.takeIf { it !== SUN }
      ?.takeIf { h.getAngle(it , SUN) > 8.5 && h.getAngle(it , SUN) <= 17.0 }
      ?.let { "comment" to arrayOf<Any>(planet)}
  }
}
